tests/stdlib/tonce.nim

```diff
@@ -0,0 +1,34 @@
+discard """
```

Comment:
js needs to be supported because `system.once` already works in js

(and leave a TODO for supporting VM because VM already doesn't work with system.once)

```nim
when true:
  var count = 0
  var countCT {.compileTime.} = 0
  proc fn(n: int) =
    once:
      when nimvm:
        countCT.inc
      else:
        count.inc
      echo (n,)
    if n > 1:
      fn(n-1)
  proc main() =
    fn(5)
    fn(4)
    when nimvm:
      echo countCT
      # doAssert countCT == 1 # fails
    else:
      doAssert count == 1
  static: main()
  main()
```